packages/core/src/hydration/event_replay.ts

```diff
@@ -58,13 +73,26 @@ export function withEventReplay(): Provider[] {
   return [
     {
       provide: IS_EVENT_REPLAY_ENABLED,
-      useValue: true,
+      useFactory: () => {
+        let isEnabled = true;
+        if (isPlatformBrowser()) {
+          // Note: globalThis[CONTRACT_PROPERTY] may be undefined in case Event Replay feature
+          // is enabled, but there are no events configured in this application, in which case
+          // we don't activate this feature, since there are no events to replay.
+          const appId = inject(APP_ID);
+          isEnabled = !!globalThis[CONTRACT_PROPERTY]?.[appId];
+        }
+        if (isEnabled) {
+          performanceMarkFeature('NgEventReplay');
+        }
+        return isEnabled;
+      },
     },
     {
       provide: ENVIRONMENT_INITIALIZER,
       useValue: () => {
         const injector = inject(Injector);
-        if (isGlobalEventDelegationEnabled(injector)) {
+        if (!shouldEnableEventReplay(injector)) {
```

Comment:
@iteriani quick question: do we need the logic from the `ENVIRONMENT_INITIALIZER` on the server or should we avoid invoking it like this?
```suggestion
        if (!isPlatformBrowser() || !shouldEnableEventReplay(injector)) {
```

Reply:
the only thing we need is IS_EVENT_REPLAY_ENABLED to be set to true on serverside.

Reply:
Thanks for additional info, I've updated the logic to invoke it only on the client.